pipeline/routing/ppr.py

```diff
@@ -33,56 +33,93 @@ def insertPathsElementsRef(cur, edges, path_counter):
             )
 
 
-def insertAccessSpaces(cur, osm_id, osm_type, IFOPT, tags, geom):
+def insertAccessSpaces(cur, osm_id, level, IFOPT, tags, geom):
     geomString = "POINT(" + str(geom[0]) + " " + str(geom[1]) + ")"
     try:
         # use INSERT INTO ... ON CONFLICT DO NOTHING to avoid duplicate entries
         cur.execute(
-            'INSERT INTO access_spaces (osm_id, osm_type, "IFOPT", tags, geom) VALUES (%s, %s, %s, %s, ST_GeomFromText(%s, 4326)) ON CONFLICT DO NOTHING',
-            (osm_id, osm_type, IFOPT, tags, geomString)
+            'INSERT INTO access_spaces (osm_id, osm_type, "level", "IFOPT", tags, geom) VALUES (%s, %s, %s, %s, %s, ST_GeomFromText(%s, 4326)) ON CONFLICT DO NOTHING',
+            (osm_id, 'N', level, IFOPT, tags, geomString)
         )
     except Exception as e:
         exit(e)
     return 1
 
 
-def identifyAccessSpaces(cur, edges):
+def identifyAccessSpaces(cur, edges, relation_id):
     # access spaces are identified, when there is:
     #   - 1) a transition from a edge_type to another (e.g. from 'footway' to 'elevator')
     #   - 2) a transition from a street_type to another (e.g. from a 'footway' 'stairs')
     special_edge_types = ["elevator"]
     special_street_types = ["stairs", "escalator", "moving_walkway"]
     edge_type = None
-    previous_edge_type = None
+    current_level = None # needed for the special cases
     
     edgeIter = iter(edges)
     
     # if the edge is the first edge of the path, there is no previous edge to compare to
     firstEdge = next(edgeIter)
     previous_edge_type = firstEdge["edge_type"]
     previous_street_type = firstEdge["street_type"]
+    previous_level = firstEdge["level"]
```

Comment:
I think we can reduce the number of variables by just storing `previous_edge`

In the for loop that comes later we can then define the variables when we need them. I would probably also drop some of them (like `level`) and then just access it like this

```python
            if edge_type == "elevator" or street_type == "stairs" or street_type == "escalator":
            previous_street_type = street_type
                # going into the elevator/stairs: use level from the previous edge
                current_level = previous_edge["level"]
            else:
                # normal case: use current level
                current_level = edge["level"]
```

Reply:
Yes this will clean up the code, implemented in [ba9dcbd](https://github.com/OPENER-next/osm2vdv462/commit/ba9dcbdd96b85776bc3d53e8c6226ba333b0de73)